bolt-sidecar/src/api/builder.rs

```diff
@@ -127,6 +125,11 @@ where
             Ok(res) => match res {
                 Err(builder_err) => builder_err,
                 Ok(header) => {
+                    // Clear the local payload cache if we have a successful response
+                    // By definition of `server.local_payload`, this will be `Some` IFF we have signed a local header
+                    let mut local_payload = server.local_payload.lock();
+                    *local_payload = None;
```

Comment:
since local_payload doesn't have a slot number, it was possible to get a stale block from this and result in a conflicting block hash in get_payload.